plugins/command-manager/gradle/formatting.gradle

```diff
@@ -0,0 +1,18 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+allprojects {
+    project.apply plugin: "com.diffplug.spotless"
+    spotless {
+        java {
+            target 'src/**/*.java'
+            removeUnusedImports()
+            eclipse().configFile rootProject.file('buildSrc/formatterConfig.xml')
```

Comment:
This is for the Eclipse IDE, which we don't use. I think this can be removed.

Reply:
In this case spotless reads that eclipse formatting configuration and uses it to validate the format of our code. It help us to validate that our code matches the conventions defined by OpenSearch.

Here is a comparison of the validation executed with and without that eclipse config.

<details><summary>Spotless with eclipse config</summary>

```powershell

> The following files had format violations:
      src/main/java/com/wazuh/setup/SetupPlugin.java
          @@ -1,4 +1,5 @@
           /*
          +·*·Copyright·OpenSearch·Contributors
           ·*·SPDX-License-Identifier:·Apache-2.0
           ·*
           ·*·The·OpenSearch·Contributors·require·contributions·made·to
          @@ -8,6 +9,9 @@
           package·com.wazuh.setup;
           
           import·com.wazuh.setup.index.WazuhIndices;
          +import·java.util.Collection;
          +import·java.util.List;
          +import·java.util.function.Supplier;
           import·org.opensearch.client.Client;
           import·org.opensearch.cluster.metadata.IndexNameExpressionResolver;
           import·org.opensearch.cluster.node.DiscoveryNode;
          @@ -23,10 +27,6 @@
           import·org.opensearch.threadpool.ThreadPool;
           import·org.opensearch.watcher.ResourceWatcherService;
           
          -import·java.util.Collection;
          -import·java.util.List;
          -import·java.util.function.Supplier;
          -
           /**
           ·*·Main·class·of·the·Indexer·Setup·plugin.·This·plugin·is·responsible·for
           ·*·the·creation·of·the·index·templates·and·indices·required·by·Wazuh·to
          @@ -39,22 +39,21 @@
           ····/**
           ·····*·Default·constructor
           ·····*/
          -····public·SetupPlugin()·{
          -····}
          +····public·SetupPlugin()·{}
           
           ····@Override
           ····public·Collection<Object>·createComponents(
          -············Client·client,
          -············ClusterService·clusterService,
          -············ThreadPool·threadPool,
          -············ResourceWatcherService·resourceWatcherService,
          -············ScriptService·scriptService,
          -············NamedXContentRegistry·xContentRegistry,
          -············Environment·environment,
          -············NodeEnvironment·nodeEnvironment,
          -············NamedWriteableRegistry·namedWriteableRegistry,
          -············IndexNameExpressionResolver·indexNameExpressionResolver,
          -············Supplier<RepositoriesService>·repositoriesServiceSupplier
          +········Client·client,
      ... (13 more lines that didn't fit)
  Run './gradlew :spotlessApply' to fix these violations.

``` 
</details>

<details><summary>Spotless without eclipse config</summary>

```powershell
> The following files had format violations:
      src/main/java/com/wazuh/setup/SetupPlugin.java
          @@ -1,4 +1,5 @@
           /*
          +·*·Copyright·OpenSearch·Contributors
           ·*·SPDX-License-Identifier:·Apache-2.0
           ·*
           ·*·The·OpenSearch·Contributors·require·contributions·made·to
          @@ -8,6 +9,9 @@
           package·com.wazuh.setup;
           
           import·com.wazuh.setup.index.WazuhIndices;
          +import·java.util.Collection;
          +import·java.util.List;
          +import·java.util.function.Supplier;
           import·org.opensearch.client.Client;
           import·org.opensearch.cluster.metadata.IndexNameExpressionResolver;
           import·org.opensearch.cluster.node.DiscoveryNode;
          @@ -23,10 +27,6 @@
           import·org.opensearch.threadpool.ThreadPool;
           import·org.opensearch.watcher.ResourceWatcherService;
           
          -import·java.util.Collection;
          -import·java.util.List;
          -import·java.util.function.Supplier;
          -
           /**
           ·*·Main·class·of·the·Indexer·Setup·plugin.·This·plugin·is·responsible·for
           ·*·the·creation·of·the·index·templates·and·indices·required·by·Wazuh·to
  Run './gradlew :spotlessApply' to fix these violations.
```

Reply:
Anyway it could be replaced by the googleJavaFormat options referenced [here](https://github.com/opensearch-project/performance-analyzer/blob/1357d86e3936c1de8de951e85f2db792e49d8a02/build.gradle#L95)